discovery-provider/src/queries/notifications.py

```diff
@@ -433,6 +434,46 @@ def notifications():
             const.playlists
         ] = playlist_favorite_dict
 
+        #
+        # Query relevant tier change information
+        #
+        balance_change_query = session.query(UserBalanceChange)
+
+        # Impose min block number restriction
+        balance_change_query = balance_change_query.filter(
+            UserBalanceChange.blocknumber > min_block_number,
+            UserBalanceChange.blocknumber <= max_block_number,
+        )
+
+        balance_change_results = balance_change_query.all()
+        tier_change_notifications = []
+
+        for entry in balance_change_results:
+            # Check for a tier change and add to tier_change_notification
+            tier = None
+            if entry.previous_balance < 100000 <= entry.current_balance:
```

Comment:
seems we're comparing strings (previous_balance and current_balance) to a number